File: Code/Cplus/LAB07_01/LAB07_01.cpp
#include <iostream>
using namespace std;

// ฟังก์ชันสำหรับสลับค่าของตัวแปรสองตัวโดยใช้พอยน์เตอร์
void swapValues(int* a, int* b) {
    int temp = *a;  // เก็บค่าของตัวแปร a ไว้ในตัวแปรชั่วคราว
    *a = *b;        // เปลี่ยนค่าของ a ให้เท่ากับค่าของ b
    *b = temp;      // เปลี่ยนค่าของ b ให้เท่ากับค่าชั่วคราว
}

int main() {
    int x, y;

    // รับค่าจากผู้ใช้
    cout << "Enter value for x: ";
    cin >> x;
    cout << "Enter value for y: ";
    cin >> y;

    // แสดงค่าก่อนการสลับ
    cout << "\nBefore swapping:" << endl;
    cout << "x = " << x << ", y = " << y << endl;

    // เรียกฟังก์ชันเพื่อสลับค่า
    swapValues(&x, &y);

    // แสดงค่าหลังการสลับ
    cout << "\nAfter swapping:" << endl;
    cout << "x = " << x << ", y = " << y << endl;

    return 0;
}

File: Code/Python/LAB07_01.py
# ฟังก์ชันสำหรับสลับค่าของตัวแปรสองตัว
def swap_values(a, b):
    return b, a

def main():
    # รับค่าจากผู้ใช้
    x = int(input("Enter value for x: "))
    y = int(input("Enter value for y: "))

    # แสดงค่าก่อนการสลับ
    print("\nBefore swapping:")
    print(f"x = {x}, y = {y}")

    # เรียกฟังก์ชันเพื่อสลับค่า
    x, y = swap_values(x, y)

    # แสดงค่าหลังการสลับ
    print("\nAfter swapping:")
    print(f"x = {x}, y = {y}")

if __name__ == "__main__":
    main()
